### PR TITLE
chore: several UI improvement to `storage init`

### PR DESCRIPTION
**Pull request by Lou1415926**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/storage_init.go [hunk 1]
```diff
@@ -1266,12 +1263,16 @@ Resource names are injected into your containers as environment variables for ea
 		auroraFlagSet.AddFlag(cmd.Flags().Lookup(f))
 	}
 
+	othersFlagSet := pflag.NewFlagSet("Others", pflag.ContinueOnError)
+	othersFlagSet.AddFlag(cmd.Flags().Lookup(storageAddIngressFromFlag))
+
 	cmd.Annotations = map[string]string{
 		// The order of the sections we want to display.
-		"sections":          `Required,DynamoDB,Aurora Serverless`,
+		"sections":          `Required,DynamoDB,Aurora Serverless,Others`,
```
Comment (efekarakus) [inline internal/pkg/cli/storage_init.go]: nit: maybe?
```suggestion
		"sections":          `Required,DynamoDB,Aurora Serverless,Optional`,
```

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/cli/storage_init.go (hunk 1 above)
Comment (Lou1415926) [reply internal/pkg/cli/storage_init.go]: done!

also I added this commit after your approval: https://github.com/aws/copilot-cli/pull/4518/commits/1a80aa98f7ab3d88d25a23b6f8349adbdff4b00d 😰

File: internal/pkg/cli/flag.go [hunk 2]
```diff
@@ -176,6 +177,16 @@ Cannot be specified with --%s.`, imageFlag)
 Cannot be specified with --%s.`, imageFlag)
 	storageTypeFlagDescription = fmt.Sprintf(`Type of storage to add. Must be one of:
 %s.`, strings.Join(applyAll(storageTypes, strconv.Quote), ", "))
+	storageLifecycleFlagDescription = fmt.Sprintf(`Whether the storage should be created and deleted
+at the same time as an workload or an environment.
```
Comment (huanjani) [inline internal/pkg/cli/flag.go]:
```suggestion
at the same time as a workload or an environment.
```

File: internal/pkg/cli/flag.go [hunk 3]
```diff
@@ -176,6 +177,16 @@ Cannot be specified with --%s.`, imageFlag)
 Cannot be specified with --%s.`, imageFlag)
 	storageTypeFlagDescription = fmt.Sprintf(`Type of storage to add. Must be one of:
 %s.`, strings.Join(applyAll(storageTypes, strconv.Quote), ", "))
+	storageLifecycleFlagDescription = fmt.Sprintf(`Whether the storage should be created and deleted
```
Comment (huanjani) [inline internal/pkg/cli/flag.go:180]:
```suggestion
	storageLifecycleFlagDescription = fmt.Sprintf(`Whether the storage resource should be created and deleted
```

File: internal/pkg/cli/flag.go [hunk 4]
```diff
@@ -176,6 +177,16 @@ Cannot be specified with --%s.`, imageFlag)
...
+at the same time as an workload or an environment.
+Must be one of: %s.`, english.OxfordWordSeries(applyAll(validLifecycleOptions, strconv.Quote), "or"))
+	storageAddIngressFromFlagDescription = fmt.Sprintf(`The workload that needs access to an
+environment storage. Must be specified with
```
Comment (huanjani) [inline internal/pkg/cli/flag.go]:
```suggestion
environment storage resource. Must be specified with
```

File: internal/pkg/cli/flag.go [hunk 5]
```diff
@@ -287,12 +298,8 @@ Uploaded asset locations are filled in the template configuration.`
...
 
 	// Storage.
-	storageFlagDescription               = "Name of the storage resource to create."
-	storageWorkloadFlagDescription       = "Name of the service or job to associate with storage."
-	storageLifecycleFlagDescription      = "Whether the storage should be created and deleted at the same time as a workload or as the environment"
-	storageAddIngressFromFlagDescription = `The workload that needs access to an environment-level addon.
-Must be specified with --name and --storage-type.
-Can be specified with --engine.`
+	storageFlagDescription             = "Name of the storage resource to create."
+	storageWorkloadFlagDescription     = "Name of the service/job that access the storage."
```
Comment (huanjani) [inline internal/pkg/cli/flag.go]:
```suggestion
	storageWorkloadFlagDescription     = "Name of the service/job that accesses the storage resource."
```

File: internal/pkg/cli/flag.go (hunk 3 above)
Comment (Lou1415926) [reply internal/pkg/cli/flag.go:180]: actually, is it important that we have "resource" 😢 ? without the resource, the help menu renders so nicely

Comment (Lou1415926) [reply internal/pkg/cli/flag.go:180]: Screenshot 2023-02-17 at 3 50 44 PM
that all 3 of these rows are filled, and the newline cuts into the nice places.

File: internal/pkg/cli/flag.go (hunk 4 above)
Comment (Lou1415926) [reply internal/pkg/cli/flag.go]: Screenshot 2023-02-17 at 3 54 18 PM

same here 🥲

File: internal/pkg/cli/flag.go (hunk 5 above)
Comment (Lou1415926) [reply internal/pkg/cli/flag.go]: Screenshot 2023-02-17 at 3 55 40 PM

and same here 🥲

Comment (Lou1415926) [reply internal/pkg/cli/flag.go]: with resource,
Screenshot 2023-02-17 at 3 56 09 PM

File: internal/pkg/cli/flag.go (hunk 3 above)
Comment (huanjani) [reply internal/pkg/cli/flag.go:180]: Nah, it's fine without, I think. :)

File: internal/pkg/cli/flag.go (hunk 4 above)
Comment (huanjani) [reply internal/pkg/cli/flag.go]: I still think it's a little weird, but it's a tech thing: https://helpcenter.veeam.com/docs/styleguide/tw/storage.html

Comment (huanjani) [reply internal/pkg/cli/flag.go]: There is [some debate](https://english.stackexchange.com/questions/291079/is-storage-countable-or-uncountable-if-it-means-warehouse-or-repository) online if "storage" is a countable word, but I think it's fine in this menu context but let's add it in the docs (which you have)! Thank you!

**Review by huanjani — APPROVED** [review]
😻

**Pull request merged**